qiskit/transpiler/passes/optimization/light_cone.py

```diff
@@ -0,0 +1,137 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2017, 2019.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Cancel the redundant (self-adjoint) gates through commutation relations."""
+from __future__ import annotations
+import warnings
+from qiskit.circuit import Gate, Qubit
+from qiskit.circuit.commutation_checker import CommutationChecker
+from qiskit.circuit.commutation_library import standard_gates_commutations
+from qiskit.circuit.library import PauliGate, ZGate
+from qiskit.dagcircuit import DAGCircuit
+from qiskit.transpiler.basepasses import TransformationPass
+from qiskit.transpiler.passes.utils.remove_final_measurements import calc_final_ops
+
+commutator = CommutationChecker(standard_gates_commutations)
```

Comment:
Could you use the `SessionCommutationChecker` here (same import path as `CommutationChecker`)? That way we can save the initialization and re-use already cached relations

Reply:
Yes